src/layman/layer/rest_workspace_layer.py

```diff
@@ -102,31 +96,23 @@ def patch(workspace, layername):
         kwargs['style_type'] = style_type
         kwargs['store_in_geoserver'] = style_type.store_in_geoserver
         delete_from = 'layman.layer.qgis.wms'
-    if len(files) > 0:
+    if len(input_files.raw_paths) > 0:
         delete_from = 'layman.layer.filesystem.input_file'
 
     # FILE NAMES
-    filenames = None
+    use_chunk_upload = bool(input_files.sent_paths)
     if delete_from == 'layman.layer.filesystem.input_file':
-        if use_chunk_upload and zipped_file:
-            filenames = list()
-        elif use_chunk_upload:
-            filenames = files
-        elif zipped_file:
-            filenames = fs_util.get_filenames_from_zip_storage(zipped_file, with_zip_in_path=True)
-        else:
-            filenames = [f.filename for f in files]
-        if not (use_chunk_upload and zipped_file):
-            input_file.check_filenames(workspace, layername, filenames,
+        if not (use_chunk_upload and input_files.is_one_archive):
+            input_file.check_filenames(workspace, layername, input_files,
                                        check_crs, ignore_existing_files=True)
         # file checks
         if not use_chunk_upload:
             temp_dir = tempfile.mkdtemp(prefix="layman_")
-            input_file.save_layer_files(workspace, layername, files, check_crs, output_dir=temp_dir,
-                                        zipped=bool(zipped_file))
+            input_file.save_layer_files(workspace, layername, input_files.sent_streams, check_crs,
+                                        output_dir=temp_dir, zipped=input_files.is_one_archive)
 
-    if filenames:
-        file_type = input_file.get_file_type(input_file.get_main_file_name(filenames))
+    if input_files.raw_paths:
+        file_type = input_file.get_file_type(input_file.get_main_file_name(input_files.raw_or_archived_paths))
```

Comment:
I think, you can use directly `input_files.raw_or_archived_main_file_path` instead of `input_file.get_main_file_name(input_files.raw_or_archived_paths)`